packages/device_info_plus/device_info_plus/lib/src/model/android_device_info.dart

```diff
@@ -3,14 +3,14 @@
 // found in the LICENSE file.
 
 import 'dart:math' as math show sqrt;
-import 'base_device_info.dart';
+import 'package:device_info_plus_platform_interface/model/base_device_info.dart';
 
 /// Information derived from `android.os.Build`.
 ///
 /// See: https://developer.android.com/reference/android/os/Build.html
-class AndroidDeviceInfo implements BaseDeviceInfo {
-  /// Android device Info class.
-  AndroidDeviceInfo({
+class AndroidDeviceInfo extends BaseDeviceInfo {
+  AndroidDeviceInfo._({
+    required Map<String, dynamic> data,
```

Comment:
This private constructor and the `fromMap` method could be refactored further, probably, but I didn't do it yet. At least I marked this constructor as private now, which shouldn't be used outside the plugin.